src/impl_float_maths.rs

```diff
@@ -0,0 +1,157 @@
+        fn is_nan
+        /// Return `true` if any element is `NaN` (not a number).
+        fn is_nan_any -> is_nan
+
+        /// If the number is infinity, then `true` is returned for each element.
+        fn is_infinite
+        /// Return `true` if any element is infinity.
+        fn is_infinite_any -> is_infinite
+    }
+    unary_op! {
+        /// The largest integer less than or equal to each element.
+        fn floor
+
+        /// The smallest integer less than or equal to each element.
+        fn ceil
+
+        /// The nearest integer of each element.
+        fn round
+
+        /// The integer part of each element.
+        fn trunc
+
+        /// The fractional part of each element.
+        fn fract
+
+        /// Absolute of each element.
+        fn abs
+
+        /// Sign number of each element.
+        ///
+        /// + `1.0` for all positive numbers.
+        /// + `-1.0` for all negative numbers.
+        /// + `NaN` for all `NaN` (not a number).
+        fn signum
+
+        /// The reciprocal (inverse) of each element, `1/x`.
+        fn recip
+
+        /// Square root of each element.
+        fn sqrt
+
+        /// `e^x` of each element. (Exponential function)
+        fn exp
+
+        /// `2^x` of each element.
+        fn exp2
+
+        /// Natural logarithm of each element.
+        fn ln
+
+        /// Base 2 logarithm of each element.
+        fn log2
+
+        /// Base 10 logarithm of each element.
+        fn log10
+
+        /// Cubic root of each element.
+        fn cbrt
+
+        /// Sine of each element. (in radians)
+        fn sin
+
+        /// Cosine of each element. (in radians)
+        fn cos
+
+        /// Tangent of each element. (in radians)
+        fn tan
+
+        /// Converts radians to degrees for each element.
+        fn to_degrees
+
+        /// Converts degrees to radians for each element.
+        fn to_radians
+    }
+    binary_op! {
+        /// Integer power of each element.
+        ///
+        /// This function is generally faster than using float power.
+        fn powi(i32)
+
+        /// Float power of each element.
+        fn powf(A)
+
+        /// Logarithm of each element with respect to an arbitrary base.
+        fn log(A)
+
+        /// The positive difference between given number and each element.
+        fn abs_sub(A)
+    }
+
+    /// Square of each element.
+    pub fn square(&self) -> Array<A, D> {
+        self.mapv(|v| v * v)
+    }
+
+    /// Limit the values for each element.
+    ///
+    /// ```
+    /// use ndarray::{Array1, array};
+    ///
+    /// let a = Array1::range(0., 10., 1.);
+    /// assert_eq!(a.clip(1., 8.), array![1., 1., 2., 3., 4., 5., 6., 7., 8., 8.]);
+    /// assert_eq!(a.clip(8., 1.), array![1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]);
+    /// assert_eq!(a.clip(3., 6.), array![3., 3., 3., 3., 4., 5., 6., 6., 6., 6.]);
+    /// ```
+    pub fn clip(&self, min: A, max: A) -> Array<A, D> {
```

Comment:
I'd hold off including this until `num-traits` adds a `clamp` method to `Float`. In particular, the behavior should be the same as `std`'s `clamp` methods (on `f32`, `f64`, and `Ord`) in the presence of NaNs and the `min > max` case.

Reply:
Performance is also an issue and the current implementation would probably need tweaking